cwrc_common.rb

```diff
@@ -85,12 +85,13 @@ def self.download_cwrc_obj(cookie, cwrc_obj, cwrc_file)
           # to detect transport corruption
           raise CWRCArchivingError unless response['CWRC-CHECKSUM'].tr('"', '') == Digest::MD5.file(cwrc_file).to_s
         elsif response.is_a? Net::HTTPServerError
-          raise Net::HTTPServerError.new("Failed request #{obj_path} with http status #{response.code}", response.code)
+          raise Net::HTTPError.new("Failed request #{obj_path} with http status #{response.code}", response.code)
         else
           raise Net::HTTPError.new("Failed request #{obj_path} with http status #{response.code}", response.code)
         end
       end
-    rescue Net::ReadTimeout,
+    rescue CWRCArchivingError,
+           Net::ReadTimeout,
```

Comment:
- fix bug related to exception handling: `HTTPServerError` is not an exception.
- add `CWRCArchivingError` to the list of exceptions that yield a retry